Permission denied when building Docker image

## Problem Description

### What is actually happening

When running docker-compose I get an error that the build cannot access `/opt/safe-settings` when installing the npm packages.

I think the issue has been introduced with the recent root/user changes in the Dockerfile. In my opinion the fix is to run the build process as root and change only after the `npm install` command to the node user.

### What is the expected behavior

Building the Docker image should be successful.

### Error output, if available

```
npm WARN deprecated @types/pino-pretty@5.0.0: This is a stub types definition. pino-pretty provides its own type definitions, so you do not need this installed.
npm WARN deprecated @types/pino-std-serializers@4.0.0: This is a stub types definition. pino-std-serializers provides its own type definitions, so you do not need this installed.
npm WARN checkPermissions Missing write access to /opt/safe-settings
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^2.3.2 (node_modules/jest-haste-map/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN safe-settings@0.1.0-rc.26 No description

npm ERR! code EACCES
npm ERR! syscall access
npm ERR! path /opt/safe-settings
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, access '/opt/safe-settings'
npm ERR!  [Error: EACCES: permission denied, access '/opt/safe-settings'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/opt/safe-settings'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2022-12-13T15_59_49_444Z-debug.log
```


## Context

### Are you using the hosted instance of probot/settings or running your own?
self-hosted

### If running your own instance, are you using it with github.com or GitHub Enterprise?
GitHub.com

#### Version of probot/settings
latest

#### Version of GitHub Enterprise

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,13 +1,10 @@
 FROM node:14.17.6-alpine
+WORKDIR /opt/safe-settings
 ENV NODE_ENV production
 ## Set the Labels
 LABEL version="1.0" \
       description="Probot app which is a modified version of Settings Probot GitHub App" \
       maintainer="GitHub Professional Services <services@github.com>"
-
-USER node
-## Set our working directory
-WORKDIR /opt/safe-settings
 
 ## These files are copied separately to allow updates
 ## to the image to be as small as possible
@@ -26,13 +23,6 @@ COPY  lib /opt/safe-settings/lib
 ## using this in their enterprise environments
 #COPY --chown=node:node .ssh/safe-settings.pem /opt/safe-settings/.ssh/
 
-# We need Python for Probot
-USER root
-RUN apk add --no-cache make python
-
-## Best practice, don't run as `root`
-USER node
-
 #
 ## Not strictly necessary, but set permissions to 400
 #RUN chmod 400 /opt/safe-settings/.ssh/safe-settings.pem /opt/safe-settings/.env
@@ -42,6 +32,9 @@ RUN npm install
 
 ## This app will listen on port 3000
 EXPOSE 3000
+
+## Best practice, don't run as `root`
+USER node
 
 ## This does not start properly when using the ['npm','start'] format
 ## so stick with just calling it outright

--- a/Dockerfile
+++ b/Dockerfile
@@ -12,28 +12,12 @@ COPY  package.json /opt/safe-settings/
 COPY  index.js /opt/safe-settings/
 COPY  lib /opt/safe-settings/lib
 
-## You should edit .env.example and save it before building this image
-## Future updates to this Dockerfile _may_ move this over to
-## pure environment variables in Docker, so it can be passed at the CLI.
-## This will be purely based on demand
-##COPY  .env /opt/safe-settings/
-
-## This can probably be removed, but users will have to make sure they
-## run the container, then copy the key. This helps avoid that for folks
-## using this in their enterprise environments
-#COPY --chown=node:node .ssh/safe-settings.pem /opt/safe-settings/.ssh/
-
-#
-## Not strictly necessary, but set permissions to 400
-#RUN chmod 400 /opt/safe-settings/.ssh/safe-settings.pem /opt/safe-settings/.env
-
 ## Install the app and dependencies
 RUN npm install
 
 ## This app will listen on port 3000
 EXPOSE 3000
 
-## Best practice, don't run as `root`
 USER node
 
 ## This does not start properly when using the ['npm','start'] format

